pkg/ebpf/c/tracee.bpf.c

```diff
@@ -5500,25 +5510,15 @@ int BPF_KPROBE(trace_sock_alloc_file)
     if (!is_socket_supported(sock))
         return 0;
 
-    // initialize program data
-
-    program_data_t p = {};
-    if (!init_program_data(&p, ctx))
```

Comment:
This is fine, because now filtering only will be done at the submit point of the packet right?
But then, we could potentially have much more packets being processed in the cgroup program side. Not sure if this will overall be better for performance, and it should be benchmarked in particular (I will run this branch on my cluster manually as well, but you should confirm with your oneliner too).
This may of course be redundant as we introduce @josedonizetti's refactor, but until then...

Reply:
This saves the initialization here since in any case we will do the initialization in `trace_ret_sock_alloc_file`.
The next few lines in this function just saves the arguments of the function, which has no effect on performance (tested)